### PR TITLE
Including hub server for integration tests

### LINKED ISSUE #13: Including Hub Server for integration tests
To be able to properly run integration tests against the `dwyl/smart-home-*` stack we should include the Hub server in our test environment so we can test against it.

This has several benefits:

+ **No need to mock a hub client**. We can use our actual hub client and run tests to make sure this is functioning as expected. This avoids any errors with our Mock implementation and makes the tests less complex to maintain.

+ **Test the internal API**. By testing against our actual hub server, we have the added benefit of checking our internal websocket api still works and we haven't accidentally created any breaking changes.

+ **Integration tests across the whole stack**. We can make sure our entire stack functions as expected by running integration tests that encompass the whole stack.

----

### TODO:

+ [ ] Add Hub dependencies to travis - postgresql, dwyl_auth key etc.
+ [ ] Add hub server to travis through git(?)
+ [ ] Rewrite existing unit tests to use the real hub not our mock client
+ [ ] Write thorough integration tests across the whole stack.

### PR DESCRIPTION
**Pull request by t0mhaines**
Includes the features detailed in #13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ cache:
   directories:
   - _build
   - deps
-  - smart-home-auth-server/deps
+  - smart-home-auth-server-deps
```
Comment (nelsonic) [inline .travis.yml:13]: is this just so you can cache the deps? 💭

Comment (t0mhaines) [reply .travis.yml:13]: Yeah, couldn't get it to work however and have commented it out in 9202264

Comment (nelsonic) [reply .travis.yml:13]: Fair enough. A worthwhile endeavour if possible, but not a deal-breaker if not. 👍

**Comment by nelsonic** [comment]
@th0mas hopefully you can use the Heroku version of the Hub/Auth server https://github.com/dwyl/smart-home-auth-server/pull/19 for this. 🤞

**Comment by t0mhaines** [comment]
Fixed in an alternative way in dwyl/smart-home-auth-server#14

**Comment by nelsonic** [comment]
@th0mas a lot of the detail in this PR is not captured anywhere and you spent a decent chunk of time exploring it.
If we ever need to run multiple apps on Travis-CI again, it could be worth having this documented somewhere.
e.g: a `run-multiple-apps.md` in https://github.com/dwyl/learn-travis 💭

**Pull request closed without merging**